src/main/java/mx/erickb/shipping/service/ShippingService.java

```diff
@@ -1,62 +1,109 @@
 package mx.erickb.shipping.service;
 
 import mx.erickb.shipping.amqp.RabbitMqSender;
-import mx.erickb.shipping.exception.InvalidResponseException;
+import mx.erickb.shipping.exception.InvalidRequestException;
+import mx.erickb.shipping.exception.NotFoundException;
 import mx.erickb.shipping.model.*;
 import mx.erickb.shipping.util.ResponseMapper;
+import mx.erickb.shipping.util.RouteUtils;
+import mx.erickb.shipping.util.RoutesGraph;
 import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Service;
 
+import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 
 @Service
 public class ShippingService implements IShippingService {
+    private static final String TYPE_VELOCITY = "transportVelocity";
+    private static final String TYPE_PACKAGE = "packageType";
+    private static final String TYPE_SIZE = "packageSizeByType";
+    private static final String TYPE_TRANSPORT = "transportType";
+    private static final String TYPE_CITY = "city";
+    private static final String TYPE_ROUTE = "routesList";
+
+    private final Logger logger = LoggerFactory.getLogger(ShippingService.class);
     private final RabbitMqSender sender;
     private final ResponseMapper mapper;
+    private final RouteUtils utils;
 
-    public ShippingService(final RabbitMqSender sender, final ResponseMapper mapper) {
+    public ShippingService(final RabbitMqSender sender, final ResponseMapper mapper, final RouteUtils utils) {
```

Comment:
Is there a reason to use final keyword here?.. I don't think they provide any value